main/streams/plain_wrapper.c

```diff
@@ -577,9 +577,11 @@ static int php_stdiop_stat(php_stream *stream, php_stream_statbuf *ssb)
 	php_stdio_stream_data *data = (php_stdio_stream_data*) stream->abstract;
 
 	assert(data != NULL);
+	if(ret = do_fstat(data, 1) == 0) {
+		memcpy(&ssb->sb, &data->sb, sizeof(ssb->sb));
+		return ret;
+	}
 
-	ret = do_fstat(data, 1);
-	memcpy(&ssb->sb, &data->sb, sizeof(ssb->sb));
 	return ret;
```

Comment:
ok, so you return whatever `do_fstat` returns (which means we could delete lilne 582, right?